### PR TITLE
chore(rolldown): oxc v0.40.1

### PR DESCRIPTION
**Pull request by Boshen**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -171,13 +171,14 @@ xxhash-rust         = "0.8.10"
...
+oxc_transform_napi = { version = "0.40.0" }
 
 # oxc crates in their own repos
-oxc_index     = { version = "1.0.0", features = ["rayon"] } # https://github.com/oxc-project/oxc-index-vec
-oxc_resolver  = { version = "2.1.1" }                       # https://github.com/oxc-project/oxc-resolver
-oxc_sourcemap = { version = "1.0.3" }                       # https://github.com/oxc-project/oxc-sourcemap
+# Versions must be relaxed for usage in oxc.
+oxc_index     = { version = "2", features = ["rayon"] } # https://github.com/oxc-project/oxc-index-vec
+oxc_resolver  = { version = "2.1.1" }                   # https://github.com/oxc-project/oxc-resolver
+oxc_sourcemap = { version = "1" }                       # https://github.com/oxc-project/oxc-sourcemap
```
Comment (hyf0) [inline Cargo.toml:181]:
```suggestion
oxc_sourcemap = { version = "1.0.3" }                       # https://github.com/oxc-project/oxc-sourcemap
```

Comment (Boshen) [reply Cargo.toml:181]: We get problematic dependency hell when I use `string_wizard` inside oxc. A upgrade to `oxc_sourcemap` breaks both repos :-/

Comment (Boshen) [reply Cargo.toml:181]: So relaxed versions in Cargo.toml, but upgrade Cargo.lock when packages upgrade.

**Comment by Boshen** [comment]
Why didn't it break when I used the git version?

File: packages/rolldown/src/binding.d.ts [hunk 2]
```diff
@@ -626,7 +652,7 @@ export interface IsolatedDeclarationsOptions {
 export interface IsolatedDeclarationsResult {
   code: string
   map?: SourceMap
-  errors: Array<string>
+  errors: Array<Error>
```
Comment (hyf0) [inline packages/rolldown/src/binding.d.ts]: Here is wrong I guess. https://github.com/rolldown/rolldown/pull/3104#issuecomment-2532213868

**Comment by Boshen** [comment]
@hyf0 can you make a new `string_wizard` release once this is merged?

**Comment by hyf0** [comment]
> @hyf0 can you make a new `string_wizard` release once this is merged?

sure

**Review by hyf0 — APPROVED** [review]

**Comment by hyf0** [comment]
> @hyf0 can you make a new `string_wizard` release once this is merged?

Done `string_wizard@0.0.24`.

**Pull request merged**